streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSessionStore.java

```diff
@@ -61,14 +61,21 @@ public void init(final ProcessorContext context, final StateStore root) {
 
     @Override
     public KeyValueIterator<Windowed<K>, AGG> findSessions(final K key, final long earliestSessionEndTime, final long latestSessionStartTime) {
-        return findSessions(key, key, earliestSessionEndTime, latestSessionStartTime);
+        final KeyValueIterator<Bytes, byte[]> bytesIterator = bytesStore.fetch(
```

Comment:
This is the fix for 1).

Reply:
Maybe saving someone else from having to dig down and figure out why it's different...

Because of the way the keys are formatted, when we know we're looking for this exact key, we can search essentially from the key `[my-key , start time]` through `[my-key , end time]`, whereas the other method we were delegating to has  to scan from `[start-key]` to `[end-key]`, which (because of the way the keys are structured) can't also bound the start and end time, but has to iterate over all the rocks keys and filter out the ones that fall outside the desired time bound.

Therefore, this change can significantly tighten the range that actually gets scanned in Rocks.